### PR TITLE
Add integration tests for rate limiter

### PR DESCRIPTION
**Pull request by likebreath**
It covers rate limiter on both `block` (single-queue, throughput and IOPS) and `net` (single-queue, rx/tx throughput), which reproduces the results reported from #2333 and #2401.

Signed-off-by: Bo Chen <chen.bo@intel.com>

### CODE REVIEW

**Comment by likebreath** [comment]
Mark PR as draft to wait for the metrics CI to be re-enabled and the rate limiter tests will be run on that same physical machine.

**Comment by sboeuf** [comment]
@likebreath the metrics CI is back online, please rebase this PR and run it against the CI.

**Comment by likebreath** [comment]
@sboeuf @rbradford PTAL. I will remove the last debug commit and undraft the PR after your approvals.

For the sake of simpler CI pipeline, I added the rate-limiter tests as a new stage for the metrics CI worker directly. The debug run shows it was finished successfully within 2 mins: https://cloud-hypervisor-jenkins.westus.cloudapp.azure.com/blue/organizations/jenkins/cloud-hypervisor/detail/PR-4544/3/pipeline/317.

File: Jenkinsfile
Comment (rbradford) [inline Jenkinsfile:395]: This isn't right. You should create a new top-level stage but asking for the same agent. You shouldn't overload the metrics stage.

**Review by rbradford — CHANGES_REQUESTED** [review]

File: scripts/run_integration_tests_rate_limiter.sh
Comment (sboeuf) [inline scripts/run_integration_tests_rate_limiter.sh:1]: GIven you know the `rate_limiter` tests are gonna run on the metrics machine right after the performances tests, you should not create a new script (we already have too many of them...).

**Review by sboeuf — CHANGES_REQUESTED** [review]

File: scripts/run_integration_tests_rate_limiter.sh
Comment (rbradford) [reply scripts/run_integration_tests_rate_limiter.sh:1]: Yes I agree we have too many scripts. But I think it's a bad idea to overload the meaning of an existing script. Similarly why I asked to create a new top level job even if it is going to be scheduled by Jenkins onto the same machine.

File: tests/integration.rs
Comment (likebreath) [reply tests/integration.rs:9090]: Added a comment to make that clear.

File: tests/integration.rs
Comment (likebreath) [reply tests/integration.rs]: Good catch. Looks like the performance tests have the similar issues. I will send a separate PR to fix that.

File: Jenkinsfile
Comment (likebreath) [reply Jenkinsfile:417]: Updated. I now have it running for every PR. Please let me know if you prefer to run only for the main branch.

I was hesitate to add a top-level stage given how 'small' the rate-limiter tests are. I agree overloading metrics stage was not a good idea.

Comment (rbradford) [reply Jenkinsfile:417]: Work on the bare metal machines should generally only be run on the main branch.

Comment (likebreath) [reply Jenkinsfile:418]: Updated.

**Review by rbradford — APPROVED** [review]

**Review by sboeuf — APPROVED** [review]

**Pull request merged**
